StringUtil.java

```diff
@@ -6,21 +6,66 @@
  * Input:
  */
 public class StringUtil {
-
+    
     private static String[] URLParser(String url){
-        //WRITE YOUR CODE HERE..
-        //parse the url and get all the get parameters and return an array of Strings
-        //Input: google.com/api?param1=value1&param2=value2
-        //Output: array of strings [] = { {param1:value1}, {param2:value2} }
+       String result;
+       String[] tempArray = new String[2];
+       String[] splittedUrl = url.split("[\\?\\&]");
+       String[] resultArray = new String[5];
+       for(int i = 1;i <= splittedUrl.length-1; i++ ){
+           result="";
```

Comment:
Appreciate you have accomplished all this with split() method.
You can do other String util methods like indexof("?") and then get a substring. It will give you the second half of the string.

int index = url.indexof(?); 13
String secondHalf  = url.substring(index+1); // output --> param1=value1&param2=value2

Now you can do a split on the secondHalf;

`` String[] params =  secondHalf.split("&");  output --> {"param1=value1", "param2=value2"} ``